components/unit/file-explorer/viewer/mixin-loading.js

```diff
@@ -6,19 +6,29 @@ export default {
       loading: false
     }
   },
+  watch: {
+    loading: {
+      handler(value) {
+        this.$emit('loading', value)
+      },
+      immediate: true
+    }
+  },
   methods: {
-    async setLoading(loading) {
-      this.loading = loading
-      this.$emit('loading', this.loading)
-      if (loading) {
-        // Leave some time for the browser to render
-        // what's in the pipeline.
-        // https://www.hesselinkwebdesign.nl/2019/nexttick-vs-settimeout-in-vue/
-        // TODO avoid magic number for setTimeout
-        // On my machine 200 works and 100 doesn't,
-        // but what will happen on other machines?
-        await setTimeoutPromise(200)
-      }
+    async startLoading() {
+      this.loading = true
+      // Leave some time for the browser to render
+      // what's in the pipeline.
+      // https://www.hesselinkwebdesign.nl/2019/nexttick-vs-settimeout-in-vue/
+      // TODO avoid magic number for setTimeout
+      // On my machine 200 works and 100 doesn't,
+      // but what will happen on other machines?
+
+      // small timeout to allow the navigation drawer transition to finish
+      // note: nextTick() will not work for that purpose
+      // because it only waits for the virtual DOM to update and not the page
+      // https://stackoverflow.com/a/47636157/8238129
+      await setTimeoutPromise(300)
```

Comment:
I realized I went a full circle with my fixes and so the mixin is useful mainly for the setLoading method which I renamed to startLoading since we can just do "this.loading = false" inside the components

The timeout is only needed to allow the navigation drawer transition to finish. Without it, the hourglass still works but the transition lags.